eigensystem.py

```diff
@@ -1,207 +1,231 @@
-    print("phi0:",np.shape(phi0))
-    
-    #============================#
-    # construct stability matrix #
-    #============================#
-    S_nok = stability_matrix_nok(mu_tilde, n_tilde[ir], omega_tilde[ir], Ve[ir], phi0[ir], phi1[ir])
-    S = stability_matrix(S_nok, mu_tilde, k)
-    print("S:",np.shape(S))
-
-    #=================#
-    # get eigenvalues #
-    #=================#
-    start = time.time()
-    evals = scipy.linalg.eigvals(S)
-    end = time.time()
-    print("Time elapsed:",end-start)
-    print(evals)
-
-single_file(input_filename)
+# Copyright (C) 2021 Sherwood Richers & Sam Flynn
+# This file is part of neutrino_linear_stability <https://github.com/srichers/neutrino_linear_stability>.
+#
+# neutrino_linear_stability is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# neutrino_linear_stability is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with neutrino_linear_stability.  If not, see <http://www.gnu.org/licenses/>.
+
+#======================================================================#
+
+import time
+import numpy as np
+import h5py
+import fileinput
+import sys
+import glob
+import scipy.linalg
+
+h = 6.6260755e-27 # erg s
+hbar = h/(2.*np.pi)
+c = 2.99792458e10 # cm/s
+MeV = 1.60218e-6 # erg
+eV = MeV/1e6 # erg
+GF_GeV2 = 1.1663787e-5 # GeV^-2
+GF = GF_GeV2 / (1000*MeV)**2 * (hbar*c)**3
+Mp = 1.6726219e-24
+
+# inputs #
+target_resolution = 20
+input_filename = "lotsadata/original_data_DO/15Msun_400ms_DO.h5"
+dm2 = 2.4e-3 * eV**2 # erg
+k = 1.088e-18 # erg
+ir = 254
+
+def stability_matrix_nok(mu_tilde,n_tilde,omega_tilde, Ve, phi0, phi1):
+    matrix_size = len(mu_tilde)
+
+    # self-interaction part
+    S_nok = -np.sqrt(2) * GF * np.array([[
+        (1.0 - mu_tilde[i]*mu_tilde[j])*n_tilde[j]
+        for j in range(matrix_size)] for i in range(matrix_size)])
+    print("after SI:",np.min(S_nok),np.max(S_nok))
+    
+    # vacuum and phis
+    for i in range(matrix_size):
+        S_nok[i,i] += omega_tilde[i] \
+            + Ve \
+            + phi0 \
+            - mu_tilde[i]*phi1
+    print("after vac/matter:",np.min(S_nok),np.max(S_nok))
+
+    return S_nok
+
+def stability_matrix(S_nok, mu_tilde, k):
+    matrix_size = len(mu_tilde)
+            
+    # k term
+    S = S_nok
+    for i in range(matrix_size):
+        S[i,i] += mu_tilde[i]*k
+    print("after k:",np.min(S),np.max(S))
+
+    return S
+
+def build_k_grid(n_nu,n_nubar,numb_k):
+    #multipliers set the range away from k_target that the grid should begin/end.
+    max_multiplier=100

```

Comment:
any hard-coded numbers should go in the inputs section

Reply:
I fixed this in a new commit.